server/difido-server/src/main/java/il/co/topq/report/front/rest/ExecutionResource.java

```diff
@@ -66,37 +74,64 @@ public int post(ExecutionDetails executionDetails) {
 	}
 
 	/**
-	 * Used to update that a single execution should not be active any more.
-	 * This is Irreversible.
-	 * 
-	 * @param executionIndex
-	 *            the id of the execution
-	 * @param active
-	 *            - Set to not active
-	 * @param locked
-	 *            - Set the execution to locked. Will no be deleted
+	 * Used to update that a single execution should not be active any more. This is Irreversible.
+	 * Also allows updating the execution description & comment through the metadata parameter
 	 * 
+	 * @param executionIndex - the id of the execution
+	 * @param active - Set to not active
+	 * @param locked - Set the execution to locked. Will no be deleted
+	 * @param metadataStr - String of key-value pairs to allow updating execution description, comment and possibly other parameters in the future, 
 	 */
 	@PUT
 	@Path("/{execution: [0-9]+}")
-	public void put(@PathParam("execution") int executionIndex, @QueryParam("active") Boolean active,
-			@QueryParam("locked") Boolean locked) {
-		log.debug("PUT - Upating execution with id " + executionIndex + ". to active: " + active + " and locked: "
-				+ locked);
-		final ExecutionMetadata metadata = metadataProvider.getMetadata(executionIndex);
-		if (null == metadata) {
-			log.warn("Trying to update state of execution with id " + executionIndex +" which is not exist");
+	public void put(@PathParam("execution") int executionIndex, @QueryParam("active") Boolean active, @QueryParam("locked") Boolean locked, @QueryParam("metadata") String metadataStr) {
```

Comment:
I am not sure if it wasn't better just to create two parameters, comment and description instead of splitting the executionStr.